framework/Source/BasicOperation.swift

```diff
@@ -24,6 +24,7 @@ open class BasicOperation: ImageProcessingOperation {
                     self?.maskFramebuffer?.unlock()
                     framebuffer.lock()
                     self?.maskFramebuffer = framebuffer
+                    framebuffer.unlock()
```

Comment:
I'm really not sure about this.

Reply:
`self?.maskFramebuffer?.unlock()` I think it's unlock before a new one.

The `maskFramebuffer` and `framebuffer` are probably same object.